src/components/app/data/services/enterpriseCustomerUser.js

```diff
@@ -91,6 +113,13 @@ export async function fetchEnterpriseCourseEnrollments(enterpriseId, options = {
     ...options,
   });
   const url = `${getConfig().LMS_BASE_URL}/enterprise_learner_portal/api/v1/enterprise_course_enrollments/?${queryParams.toString()}`;
-  const response = await getAuthenticatedHttpClient().get(url);
-  return camelCaseObject(response.data);
+  try {
+    const response = await getAuthenticatedHttpClient().get(url);
+    return camelCaseObject(response.data);
+  } catch (error) {
+    if (getErrorResponseStatusCode(error) === 404) {
+      return [];
+    }
```

Comment:
[informs] Ensures calling the `enterprise_course_enrollments` API endpoint when not explicitly linked to the requested enterprise customer handles the expected 404 to avoid an expected JS error being caught in the route loader.

This also has the nice benefit that staff users will now be able to properly see course about pages for unlinked enterprise customers' learner portals, which will be helpful for support/product/UX/eng.